src/mbgl/tile/vector_tile.cpp

```diff
@@ -15,9 +15,17 @@ class VectorTileLayer;
 
 using packed_iter_type = protozero::iterator_range<protozero::pbf_reader::const_uint32_iterator>;
 
+struct VectorTileLayerData {
```

Comment:
@anandthakker @jfirebaugh - Landed here in a revised effort to integrate https://github.com/mapbox/vector-tile back into mbgl. I have some questions.

1) Without `VectorTileLayerData` and is `shared_ptr<const std::string> data` member what exactly would go wrong? I figure there must be somewhere in the code that expects to create a `VectorTileLayer` and use it/cache it in such a way beyond when the underlying data would be kept alive. So keeping a shared_ptr to the data keeps it alive as long as all `VectorTileLayer` instances are alive? Can you point to the code that needs this? (I could not find it)

2) Why does the `VectorTileFeature` need a `std::shared_ptr<VectorTileLayerData> layerData` rather than just `const VectorTileLayer&`. This indicates to me that somewhere in the code expects to keep a  `VectorTileFeature` instance alive longer than the parent `VectorTileLayer`. Is this happening and can you point to the place in the code that requires is? Or is this perhaps unneeded - in that the `VectorTileLayer` will always be alive as long as the `VectorTileFeature` created from it?

 I noticed that `shared_ptr<const std::string> data` gets passed to not only `VectorTileLayerData` but also `VectorTileFeature` and